updating/updating-disconnected-cluster.adoc

```diff
@@ -1,23 +1,134 @@
+
+.Procedure
+
+. Findout if there is link:https://access.redhat.com/solutions/4583231[an update path] exists between your current version and the version you want the cluster to move.
+
+. Mirror the version images to the internal container registry.
+
+** Connect the removable media to a system connected to the internet.
+** Mirror the images and configuration manifests to a directory on the removable media.
++
+----
+$ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+----
++
+
+** Take the media to the disconnected environment and upload the images to the local container registry
++
+----
+$ oc image mirror "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+----
++
+. Use `oc` to log in to a given cluster to be upgraded.
++
+[NOTE]
+====
+If you are upgrading from earlier release than 4.5.0, you must must use the manual method of creating configmap.
+
+This is because `oc` command enhancement is not available in previous releases.
+====
++
+. Apply the the mirrored release image signature ConfigMap to the connected cluster
++
+----
+$ oc apply -f mirror/config/signature-sha256-81154f5c03294534.yaml
+----
+. Perform the upgrade
++
+----
+$ oc adm upgrade --allow-explicit-upgrade --to-image REGISTRY/REPOSITORY@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+----
+
+[id="update-disconnected-manual_method_configmap_creation"]
+== Manual method to create the image signature configmap
+
+The following commands will create and apply the image signature configmap to the cluster you want to update. 
+
+.Procedure
+
+. Find the link:https://access.redhat.com/solutions/4583231[version] you want the cluster to update.
+
+. Add the version to OCP_RELEASE_NUMBER environment variable.
++
+----
+$ export OCP_RELEASE_NUMBER=<4.y.z>
+----
++
+. Get the release image digest from link:https://quay.io/[Quay].
++
+----
+$ export DIGEST="$(oc adm release info quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE_NUMBER}-x86_64 | sed -n 's/Pull From: .*@//p')"
+----
++
+. Get the image signature from link:https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release[mirror.openshift.com] website.
++
+----
+$ export SIGNATURE_BASE64=$(curl -s "https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/$(echo $DIGEST | cut -d: -f1)=$(echo $DIGEST | cut -d: -f2)/signature-1" | base64 -w0 && echo)
+----
++
+. Get the digest algorithm.
++
+----
+export DIGEST_ALGO=$(echo $DIGEST | cut -d: -f1)
+----
++
+. Get the digest signature.
++
+----
+export DIGEST_SIGNATURE=$(echo $DIGEST | cut -d: -f2)
```

Comment:
If you're going to split out the algo and hash (formally ["the encoded portion"][1]) into variables, can we do that before `curl`ing the signature so we don't have to drop the logic inline?

And no need to shell out to replace `:` with `=`.  If we can assume [a POSIX shell][2], we can:

```console
$ DIGEST_ALGO="${DIGEST%%:*}" # remove first colon and everything after
$ DIGEST_ENCODED="${DIGEST#*:}" # remove everything up to and including the first colon
```

`${DIGEST/:/=}` would be even more compact, but is a Bashism.

[1]: https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests
[2]: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
This thread has been replaced by [this new thread][1] and can be marked resolved.

[1]: https://github.com/openshift/openshift-docs/pull/21993#discussion_r437015337